build-tools/packages/build-cli/src/commands/release/history.ts

```diff
@@ -0,0 +1,156 @@
+ */
+import { Flags } from "@oclif/core";
+import chalk from "chalk";
+import { table } from "table";
+
+import { detectBumpType } from "@fluid-tools/version-tools";
+
+import { packageSelectorFlag, releaseGroupFlag } from "../../flags";
+import {
+    ReleaseReport,
+    VersionDetails,
+    getDisplayDate,
+    getDisplayDateRelative,
+    sortVersions,
+} from "../../lib";
+import { ReleaseGroup, ReleasePackage } from "../../releaseGroups";
+import { ReleaseReportBaseCommand, ReleaseSelectionMode } from "./report";
+
+const DEFAULT_MIN_VERSION = "0.0.0";
+
+/**
+ * Prints a list of released versions of a package or release group. Releases are gathered from the git tags in repo
+ * containing the working directory.
+ *
+ * @remarks
+ * Use 'npm view' to list published packages based on the public npm registry.
+ *
+ * The number of results can be limited using the --limit argument.
+ */
+export default class ReleaseHistoryCommand<
+    T extends typeof ReleaseHistoryCommand.flags,
+> extends ReleaseReportBaseCommand<T> {
+    static description = `Prints a list of released versions of a package or release group. Releases are gathered from the git tags in repo containing the working directory.
+
+    Use 'npm view' to list published packages based on the public npm registry.
+
+    The number of results can be limited using the --limit argument.`;
+
+    static examples = [
+        {
+            description: "List all the releases of the azure release group.",
+            command: "<%= config.bin %> <%= command.id %> -g azure",
+        },
+        {
+            description: "List the 10 most recent client releases.",
+            command: "<%= config.bin %> <%= command.id %> -g client --limit 10",
+        },
+    ];
+
+    static flags = {
+        releaseGroup: releaseGroupFlag({
+            required: false,
+            exclusive: ["package"],
+        }),
+        package: packageSelectorFlag({
+            required: false,
+            exclusive: ["releaseGroup"],
+        }),
+        limit: Flags.integer({
+            char: "l",
+            description: `Limits the number of displayed releases for each release group. Results are sorted by semver, so '--limit 10' will return the 10 highest semver releases for the release group.`,
+        }),
+        ...ReleaseReportBaseCommand.flags,
+    };
+
+    static enableJsonFlag = true;
+
+    defaultMode: ReleaseSelectionMode = "date";
+    releaseGroupOrPackage: ReleaseGroup | ReleasePackage | undefined;
+
+    public async run(): Promise<{ reports: ReleaseReport[] }> {
+        this.releaseGroupOrPackage =
+            this.processedFlags.releaseGroup ?? this.processedFlags.package;
+
+        const context = await this.getContext();
+        this.releaseData = await this.collectReleaseData(
+            context,
+            this.defaultMode,
+            this.releaseGroupOrPackage,
+            false,
+        );
+        if (this.releaseData === undefined) {
```

Comment:
```suggestion
        if (!this.releaseData) {
```

Reply:
This form is to appease [typescript-eslint/strict-boolean-expressions](https://typescript-eslint.io/rules/strict-boolean-expressions).

Reply:
whoops, ok!